python/lsst/ci/prepare.py

```diff
@@ -147,37 +145,40 @@ class ProductFetcher:
     tries
         The number of times to attempt to 'fetch' a product.
     """
-    def __init__(self,
-                 build_dir: Optional[str],
-                 repos: str,
-                 repository_patterns: Optional[str] = None,
-                 dependency_module: Optional[EupsModule] = None,
-                 version_db: Optional[VersionDb] = None,
-                 no_fetch: bool = False,
-                 out=sys.stdout,
-                 tries=1):
+
+    def __init__(
+        self,
+        build_dir: str | None,
+        repos: str,
+        repository_patterns: str | None = None,
+        dependency_module: EupsModule | None = None,
+        version_db: VersionDb | None = None,
+        no_fetch: bool = False,
+        out=sys.stdout,
+        tries=1,
+    ):
         self.build_dir = os.path.abspath(build_dir) if build_dir else None
         if repository_patterns:
-            self.repository_patterns: Optional[List[str]] = repository_patterns.split('|')
+            self.repository_patterns: list[str] | None = repository_patterns.split("|")
         else:
             self.repository_patterns = None
         self.no_fetch = no_fetch
         if repos:
             if os.path.exists(repos):
-                with open(repos, 'r', encoding='utf-8') as f:
+                with open(repos, "r", encoding="utf-8") as f:
                     self.repos = yaml.safe_load(f)
             else:
-                raise Exception("YAML repos file '%s' does not exist" % repos)
+                raise Exception(f"YAML repos file {repos!r} does not exist")
```

Comment:
It's a `str`, so `!r` is not really necessary, right?  (There are some others like this, but they're maybe not as obvious.)

Reply:
I was using `%r` to automatically add the single quote for me...

Reply:
OIC.  Cute.